sycl/source/detail/queue_impl.hpp

```diff
@@ -388,6 +388,20 @@ class queue_impl {
   template <typename Param>
   typename Param::return_type get_backend_info() const;
 
+  /// Provides a hint to the backend to execute previously issued commands on
```

Comment:
If this is not a guarantee then it is useless users like OpenMP will be assuming the job is submitted and inserting barriers to add more dependent tasks.

Reply:
I'm not sure I follow. The spec of the extension simply says that eventual execution is not guaranteed by calling ext_oneapi_prod(). Whether an app uses or not this API should not affect the correctness of the program. It is merely an encouragement to increase performance. Perhaps @gmlueck can help interpret the spec and its intended usage if I'm misunderstanding!

Reply:
@lbushi25's interpretation of the spec is correct.

@RaviNarayanaswamy, I don't understand your concern with barriers and dependent tasks.  How does that relate to whether the commands submitted prior to the "prod" call are guaranteed to start executing?

The thing that "prod" does *not* guarantee is that the commands are necessarily running at the same time as the host thread.  For example, it would be invalid for the host thread to spin reading host memory waiting until the kernel wrote some value there.

Reply:
@gmlueck  when we call fflush  we want it to submit the previous commands before returning.

Reply:
@RaviNarayanaswamy But I don't think there is a need for that. The queue class itself is an abstraction that allows you to submit commands and get an event back for synchronization. From your end you can think of the kernel as submitted. The flush functionality in this PR is concerned with encouraging execution of the kernel on the device itself, a lower level than what SYCL exposes..

Reply:
@lbushi25  is this PR implementing the feature requested in [CMPLRLLVM-37616].  If so OpenMP which calls MKL which submits the kernel and returns to OpenMP.  Now OpenMP needs to make sure the command is submitted,  we cannot use SYCL events to order the scheduling in OpenMP.  
If this is not for supporting the above JIRA and just ignore my comments.

Reply:
@RaviNarayanaswamy Based on internal discussion, it seems like the prod API is indeed sufficient for your use-case. Can you confirm? 
If so, please leave a review/approval.